pkg/version/version.go

```diff
@@ -63,40 +63,41 @@ func Get() version.Info {
-	defaultReleaseInfoPrefix = "\x00_RELEASE_IMAGE_LOCATION_\x00"
-	defaultReleaseInfoLength = len(defaultReleaseInfoPadded)
+	defaultVersionPadded = "\x00_RELEASE_VERSION_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
+	defaultVersionPrefix = "\x00_RELEASE_VERSION_LOCATION_\x00"
+	defaultVersionLength = len(defaultVersionPadded)
 )
 
 // ExtractVersion() abstracts how the binary loads the default release payload. We want to lock the binary
-// to the pull spec of the payload we test it with, and since a payload contains the cli image we can't
-// know that at build time. Instead, we make it possible to replace the release string after build via a
+// to version of the payload we test it with, and since a payload contains the cli image we can't
+// know that at build time. Instead, we make it possible to replace the version string after build via a
 // known constant in the binary.  When extracting oc from release image, 'oc version' reports the payload version.
 func ExtractVersion() (version.Info, string, error) {
-	if strings.HasPrefix(defaultReleaseInfoPadded, defaultReleaseInfoPrefix) {
+	if strings.HasPrefix(defaultVersionPadded, defaultVersionPrefix) {
 		return Get(), "", nil
 	}
-	nullTerminator := strings.IndexByte(defaultReleaseInfoPadded, '\x00')
+	nullTerminator := strings.IndexByte(defaultVersionPadded, '\x00')
 	if nullTerminator == -1 {
 		// the binary has been altered, but we didn't find a null terminator within the release name constant which is an error
-		return version.Info{}, "", fmt.Errorf("release name location was replaced but without a null terminator before %d bytes", defaultReleaseInfoLength)
+		return version.Info{}, "", fmt.Errorf("release name location was replaced but without a null terminator before %d bytes", defaultVersionLength)
 	}
-	if nullTerminator > len(defaultReleaseInfoPadded) {
+	if nullTerminator > len(defaultVersionPadded) {
 		// the binary has been altered, but the null terminator is *longer* than the constant encoded in the binary
 		return version.Info{}, "", fmt.Errorf("release name location contains no null-terminator and constant is corrupted")
 	}
-	releaseName := defaultReleaseInfoPadded[:nullTerminator]
+	releaseName := defaultVersionPadded[:nullTerminator]
 	if len(releaseName) == 0 {
 		// the binary has been altered, but the replaced release name is empty which is incorrect
-		return version.Info{}, "", fmt.Errorf("release name location is empty, this binary was incorrectly generated")
+		// the oc binary will not be pinned to Release Metadata:Version
+		return version.Info{}, "", fmt.Errorf("oc version was incorrectly replaced during extract")
```

Comment:
The oc binary doesn't depend on the version being pinned to the release, but if we're meddling w/ the binary and the result isn't what is expected, it should return an error? oc works the same either way, though, so I could see return    ing a warning for these, instead.

Reply:
I can't figure this out, when I extract oc from the release image from this PR, I get an empty releaseName (match found, but len(releaseName)==0. However, when I create a release from a published release w/ new cli image (built from this PR, but locally), and extract oc from that, releaseName is the expected 4.3.0-0.ci-2019-10-23-blah - maybe the ci-op/release:latest image is in flux w/ multiple pushes/test restarts so the Release Metadata: Version is absent? not sure.

Reply:
:+1: for the warning and printing the built-in version.

Reply:
For the 2nd, let's land this and we'll check you still have a few followups to fix. Side note, have you checked with hexeditor that binary extracted from the release image build from this PR?